lib/components/tiled_component.dart

```diff
@@ -6,17 +7,76 @@ import 'package:flame/flame.dart';
 import 'package:flutter/material.dart' show Colors;
 import 'package:tiled/tiled.dart' hide Image;
 
+/// Tiled represents all flips and rotation using three possible flips: horizontal, vertical and diagonal.
+/// This class converts that representation to a simpler one, that uses one angle (with pi/2 steps) and two flips (H or V).
+class _SimpleFlips {
+  // angle (in steps of pi/2 rads), clockwise
+  final int angle;
+  // whether to flip across a central vertical axis.
+  final bool flipH;
+  // whether to flip across a central horizontal axis.
+  final bool flipV;
+
+  _SimpleFlips(this.angle, this.flipH, this.flipV);
+
+  // This is the conversion
+  factory _SimpleFlips.fromFlips(Flips flips) {
+    int angle;
+    bool flipV, flipH;
+
+    if (!flips.diagonally && !flips.vertically && !flips.horizontally) {
+      angle = 0;
+      flipV = false;
+      flipH = false;
+    } else if (!flips.diagonally && !flips.vertically && flips.horizontally) {
+      angle = 0;
+      flipV = false;
+      flipH = true;
+    } else if (!flips.diagonally && flips.vertically && !flips.horizontally) {
+      angle = 0;
+      flipV = true;
+      flipH = false;
+    } else if (!flips.diagonally && flips.vertically && flips.horizontally) {
+      angle = 2;
+      flipV = false;
+      flipH = false;
+    } else if (flips.diagonally && !flips.vertically && !flips.horizontally) {
+      angle = 1;
+      flipV = false;
+      flipH = true;
+    } else if (flips.diagonally && !flips.vertically && flips.horizontally) {
+      angle = 1;
+      flipV = false;
+      flipH = false;
+    } else if (flips.diagonally && flips.vertically && !flips.horizontally) {
+      angle = 3;
+      flipV = false;
+      flipH = false;
+    } else if (flips.diagonally && flips.vertically && flips.horizontally) {
+      angle = 1;
+      flipV = true;
+      flipH = false;
+    } else {
+      // this should be exhaustive
+      throw 'Invalid combination of booleans: $flips';
+    }
+
+    return _SimpleFlips(angle, flipH, flipV);
+  }
+}
+
 class TiledComponent extends Component {
   String filename;
   TileMap map;
   Image image;
   Map<String, Image> images = <String, Image>{};
   Future future;
   bool _loaded = false;
+  double destTileSize;
 
   static Paint paint = Paint()..color = Colors.white;
 
-  TiledComponent(this.filename) {
+  TiledComponent(this.filename, this.destTileSize) {
```

Comment:
Shouldn't we have a doc for this? I know this is pretty clear to me what this does. But once or twice people got confused about what `destTileSize` was on other features.

Reply:
I think it's pretty clear but I can def. add :)

Reply:
yeah, like I said, I think it too, but since there were already some confusion about it, I would prefere to add a simple statement, just to be sure.